### DISCUSSION TITLE
PyArt does not display any map when using "RadarDisplay"

### DISCUSSION THREAD

**Discussion by azrellie** [ARM-DOE/pyart#1219] [Q&A] [answered]
So I'm following the [tutorial](https://nexradaws.readthedocs.io/en/latest/Tutorial.html) and when I get to this part and run it (along with the other code):
![image](https://user-images.githubusercontent.com/98503938/180593924-3cf4842f-76b2-4071-b6ea-b02870c5bff8.png)
No map is generated or anything (there is no errors either, and I have all the dependencies).

What I expect to get:
![image](https://user-images.githubusercontent.com/98503938/180593969-0b2383a4-7d09-4837-aacb-2379c83439ed.png)

But upon running it (in command prompt), I get zero results. No image file generated or window pop up of the radar data.

**Comment by zssherman** [comment]
@softieeeeee Hmmm I know notebooks can be tricky with plots showing. Can you try plt.show() at the end of your code or
```
%matplotlib inline
````
in your import cell.

**Comment by azrellie** [comment]
Using `%matplotlib inline` just errors out with 'invalid syntax'.
![image](https://user-images.githubusercontent.com/98503938/180675952-e52d3a55-d3c9-468b-8d5d-19d54d713b5e.png)

And, there is no such function named 'show()'.
![image](https://user-images.githubusercontent.com/98503938/180676896-9d8b6aa9-80d6-43b2-9e88-c1d4e7c9adca.png)

**Comment by zssherman** [comment]
Can't tell, is that calling plt.show()? I would try maybe
```
%matplotlib notebook
```
What system are you using, I am using linux and plt.show() works, but not sure if it changes between OS

**Comment by azrellie** [comment]
I'm on Windows 10 64-bit (editing the file in VSCode). Maybe that is likely the reason it doesn't work.

**Comment by mgrover1** [comment]
@softieeeeee - can you try printing `results`?

```python
results = conn.download(scans[0:4], templocation)
print(results)
```

That way we can ensure that the files are downloaded, and stored locally.

If that works, then make sure to add the figure to the plotting line:

```python
fig = plt.figure(figsize=(16,12))
for i,scan in enumerate(results.iter_success(),start=1):
    ax = fig.add_subplot(2,2,i)
    radar = scan.open_pyart()
    display = pyart.graph.RadarDisplay(radar)
    display.plot('velocity', 1, ax=ax, fig=fig, title="{} {}".format(scan.radar_id,scan.scan_time))
    display.set_limits((-150, 150), (-150, 150), ax=ax)
```

Give that a try and let us know if that solves this issue.

**Reply by azrellie** [reply]
> @softieeeeee - can you try printing `results`?
> 
> ```python
> results = conn.download(scans[0:4], templocation)
> print(results)
> ```
> 
> That way we can ensure that the files are downloaded, and stored locally.
> 
> If that works, then make sure to add the figure to the plotting line:
> 
> ```python
> fig = plt.figure(figsize=(16,12))
> for i,scan in enumerate(results.iter_success(),start=1):
>     ax = fig.add_subplot(2,2,i)
>     radar = scan.open_pyart()
>     display = pyart.graph.RadarDisplay(radar)
>     display.plot('velocity', 1, ax=ax, fig=fig, title="{} {}".format(scan.radar_id,scan.scan_time))
>     display.set_limits((-150, 150), (-150, 150), ax=ax)
> ```
> 
> Give that a try and let us know if that solves this issue.

I did forget to mention that the files are being downloaded and saved locally, as seen here:
![image](https://user-images.githubusercontent.com/98503938/180783325-dbdfa019-4d62-4f43-930c-915bcf8be485.png)
However no radar image is being generated still.

I still printed the results though anyway and this is what I got:
![image](https://user-images.githubusercontent.com/98503938/180783855-1440b940-9992-43df-a0e6-ec95cbf63afe.png)

**Reply by mgrover1** [reply]
Did you replace the plotting section with the code snippet I included above? Adding in the `fig=fig` argument?

**Comment by azrellie** [comment, marked as answer]
Yes, I did replaced the code I had with what you provided me. Also small question, does the radar image get generated as an image file or a window popup?

**Reply by mgrover1** [reply]
You'll need to remove the `%matplotlib inline` line at the beginning. When calling `plt.show()` or `fig.show()`, the script returns 

```python
<ipython-input-15-6ebedab27258>:1: UserWarning: Matplotlib is currently using module://matplotlib_inline.backend_inline, which is a non-GUI backend, so cannot show the figure.
```

Which is why the image is not popping up. You need to remove that line, or run this in a notebook (which is what the original example was written in, and why they added the inline call at the beginning).

**Reply by azrellie** [reply]
I never had `%matplotlib inline` at the beginning though. I removed it because it was causing errors via 'invalid syntax'.
This is the current code I have:
![image](https://user-images.githubusercontent.com/98503938/180786752-eae211cf-a6a9-411c-b46d-94cdf43d19c3.png)

**Reply by mgrover1** [reply]
You'll need to add a plt.show() line outside of the for loop at the end. That should fix it.

**Reply by mgrover1** [reply]
@softieeeeee did this solve the issue?

**Reply by azrellie** [reply]
Excuse me for being a little late. Was outside observing weather. But yes, it does work now. I still have a lot of questions though and how I can integrate this into a C++/C# application. Oh well, I appreciate your help though.

**Reply by mgrover1** [reply]
No worries - just wanted to make sure. Not sure if we will be able to help on the C++/C# side of things, but glad this resolved the issue!
